pkg/jobWorker/job.go

```diff
@@ -0,0 +1,374 @@
+package jobWorker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	ns "github.com/P-A-R-U-S/Go-Job-Worker-Service/pkg/jobWorker/namespaces"
+	"github.com/google/uuid"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+)
+
+var (
+	ErrJobAlreadyStarted       = errors.New("job already started")
+	ErrJobNotStarted           = errors.New("job not started")
+	ErrInvalidCommand          = errors.New("command must be provided")
+	ErrInvalidCPU              = errors.New("CPU must be greater than 0")
+	ErrInvalidIOBytesPerSecond = errors.New("IOBytesPerSecond must be greater than 0")
+	ErrInvalidMemBytes         = errors.New("MemBytes must be greater than 0")
+)
+
+type State string
+
+const (
+	JOB_STATUS_NOT_STARTED State = "NotStarted"
+	JOB_STATUS_RUNNING     State = "Running"
+	JOB_STATUS_COMPLETED   State = "Completed"
+	JOB_STATUS_TERMINATED  State = "Terminated"
```

Comment:
nit: https://google.github.io/styleguide/go/decisions#constant-names

Reply:
@smallinsky That from Java world I think - in Oracle they forced us to names constant like _(should be all uppercase with words separated by underscores ("_"))_: https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

Changed to team preferred format :))

<img width="1317" alt="Screenshot 2024-09-05 at 11 45 07 AM" src="https://github.com/user-attachments/assets/9e34785e-38a1-4222-bf25-269829b4fe49">